Shawn/Shawn_BuyMeALaptop_Challenge.py

```diff
@@ -0,0 +1,117 @@
+                    "ram": ["8GB"],
+                    "storage": ["256GB SSD","512 GB SSD", "1 TB SSD"],
+                    "colors": ["Platinum silver exterior, black interior", "Platinum silver exterior, black interior"],
+                    "price": [999, 1099, 1149, 1249]
+                },
+                {
+                    "id": "4",
+                    "screen_size": "15-inch",
+                    "cpu": ["10th Generation Intel Core i5-10300H"],
+                    "ram": ["8GB", "16GB", "32GB", "64GB"],
+                    "storage": ["256 GB SSD", "512 GB SSD", "1 TB SSD", "2 TB SSD"],
+                    "colors": ["Frost White with White Palmrest", "Frost White with White Palmrest"],
+                    "price": [1199, 1299, 1399, 1749, 1999, 2299]
+                }
+            ],
+            "description": "The Dell XPS is an absolutely brilliant laptop. The 2020  version rocks an 11th-generation Intel Core i3, i5 or i7 processor and a bezel-less Infinity Edge display. This Dell XPS continues to be the most popular Windows laptop in the world. What is more, there is a wide range of customization options, so you can really make the Dell XPS the best laptop for your needs. "
+        }
+]
+
+# TODO: Write code to print out the MacBook Pro url 
+print('Apple Macbook Pro url')
+for laptop in laptops:
+    print(laptop['url'])
+print()
+
+# TODO: Write code to print all possible prices of the 16-inch MacBook Pro.
+MacBook_Pro_prices= laptops[0]['types'][1]['price']
+for price in MacBook_Pro_prices:
+    print(price)
+print()
+
+# TODO: Write code to print all the color options for Dell XPS 13.
+color_options= laptops[1]['types'][0]['colors']
+print(color_options)
+print()
+
+# TODO: Write code to print the description of Dell XPS laptop.
+print('The Dell XPS is an absolutely brilliant laptop. The 2020  version rocks an 11th-generation Intel Core i3, i5 or i7 processor and a bezel-less Infinity Edge display. This Dell XPS continues to be the most popular Windows laptop in the world. What is more, there is a wide range of customization options, so you can really make the Dell XPS the best laptop for your needs. ')
+print()
+
+#print("Question 2: Out of Stock laptops")
+#print("Suppose that the 13-inch MacBook Pro in space gray color is sold out. Also, the same laptop with 1 TB storage is out of stock as well. Update the list of dictionaries such that these options are removed. Print the updated dictionary.
+
+# TODO: Update the laptops dictionary.
+# TODO: Print the new dictionary.
+laptops[0]['types'][0]['colors'].remove('space gray')
+laptops[1]['types'][1]['storage'].remove('1 TB SSD')
+print(laptops)
+print()
+
+
+
+#print("Question 3: listing all the prices")
+#print("Time to look at the range of prices. Using a nested loops, print out all possible computer prices")
+
+# TODO: print out all possible prices for the laptops
```

Comment:
This one is a little challenging. You had the right idea, the proper way to set up the for loop is:

`for product in laptops:
    for product_type in product['types']:
        for price in product_type['price']:
            print(price)`